### ISSUE TITLE
Fatal exceptions in tests are ignored

### ISSUE THREAD

**Issue by Arthurm1** [scalameta/metals#5379]
### Describe the bug

Create the following munit test...
```scala
class IllegalAccess extends munit.FunSuite {
    test("Handle IllegalAccessError") {
        throw new IllegalAccessError("This should cause test to fail")
    }
}
```

Run it under Metals.

Result is...
```
IllegalAccess:
Execution took 0ms
No test suite was run
```

Instead of reporting a failed test, the test is ignored and the suite completes semi-successfully.  It also causes any other tests in the suite to be ignored so it's hard to realise that the tests are failing.

This exception is thrown if `--add-exports` is missing from code that requires it.

[Here](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalAccessError.html) you can see that `IllegalAccessError` is a subclass of `java.lang.LinkageError`.
[Here](https://github.com/scala/scala/blob/93d6281876ae53d7eb0d1f1e8369437f0a260015/src/library/scala/util/control/NonFatal.scala#L42) you can see that `java.lang.LinkageError` is treated as fatal.
[Here](https://github.com/scalameta/munit/blob/854b8495fab4255c14c97378fd2a59eaf0b2830d/munit/shared/src/main/scala/munit/FunSuite.scala#L39), Munit only handles NonFatal errors.
[Here](https://github.com/junit-team/junit4/blob/16228f3ccea3c6f1170488e0e268f3601d130f75/src/main/java/org/junit/runners/ParentRunner.java#L419) JUnit runner does seem to report the error.

I'd add a stacktrace but there isn't one printed as it silently fails.

I'm not sure if this is an issue with Bloop, Metals, MUnit, JUnit or the way Scala classifies NonFatal errors.

### Expected behavior

I think this type of error should be reported and the test should gracefully fail.  It's not like we're out of memory.

### Operating system

Windows

### Editor/Extension

VS Code

### Version of Metals

v0.11.12

**Comment by filipwiech** [comment]
Coult this be connected with https://github.com/scalameta/munit/issues/650? :thinking:

**Comment by tgodzik** [comment]
Thanks for reporting! I think the issue might lie with munit. Not sure where the tests are invoked, but I would bet that is the reason, since we don't get a failure to report

**Comment by Arthurm1** [comment]
This was fixed in https://github.com/scalameta/munit/pull/669 and released in [v1.0.0-M9](https://github.com/scalameta/munit/releases/tag/v1.0.0-M9) and Metals is now on v1.0.0-M11

Quick test shows it works in Metals.
